crates/ruma-client-api/src/sync/sliding_sync_events.rs

```diff
@@ -0,0 +1,310 @@
+pub struct RoomSubscription {
+    /// Sticky. Required state for each room returned. An array of event type and state key tuples.
+    /// Note that elements of this array are NOT sticky so they must be specified in full when they
+    /// are changed.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub required_state: Option<Vec<(RoomEventType, String)>>,
+
+    /// Sticky. The maximum number of timeline events to return per room
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timeline_limit: Option<UInt>,
+}
+
+/// Sliding Sync Request
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct SyncRequest {
+    /// The lists of rooms we're interested in
+    pub lists: Vec<Raw<SyncRequestList>>,
+
+    /// Specific rooms and event types that we want to receive events from
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub room_subscriptions: Option<BTreeMap<OwnedRoomId, RoomSubscription>>,
+
+    /// Specific rooms we no longer want to receive events from
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub unsubscribe_rooms: Option<Vec<OwnedRoomId>>,
+}
+
+ruma_api! {
+    metadata: {
+        description: "Get all new events in a sliding window of rooms since the last sync or a given point of time.",
+        method: POST,
+        name: "sync",
+        added: 1.0,
+        stable_path: "/_matrix/client/v3/sync",
+        // unstable_path: "/_matrix/client/unstable/org.matrix.msc3575/sync",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    #[derive(Default)]
+    request: {
+        /// A point in time to continue a sync from.
+        ///
+        /// Should be a token from the `pos` field of a previous `/sync`
+        /// response.
+        #[serde(skip_serializing_if = "Option::is_none")]
+        #[ruma_api(query)]
+        pub pos: Option<&'a str>,
+
+        /// The maximum time to poll in milliseconds before returning this request.
+        #[serde(
+            with = "opt_ms",
+            default,
+            skip_serializing_if = "Option::is_none",
+        )]
+        #[ruma_api(query)]
+        pub timeout: Option<Duration>,
+
+        /// The sync request body to send
+        #[ruma_api(body)]
+        pub body: SyncRequest,
+
+    }
+
+    response: {
+        /// Present and true if this response describes an initial sync
+        /// (i.e. after the `pos` token has been discard by the server?)
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pub initial: Option<bool>,
+
+        /// The token to supply in the `pos` param of the next `/sync` request.
+        pub pos: String,
+
+        /// Updates to the sliding room list
+        #[serde()]
+        pub lists: Vec<SyncList>,
+
+        /// The updates on rooms
+        pub rooms: Option<BTreeMap<OwnedRoomId, SlidingSyncRoom>>
```

Comment:
```suggestion
        #[serde(default, skip_serializing_if = "BTreeMap::is_empty")]
        pub rooms: BTreeMap<OwnedRoomId, SlidingSyncRoom>,
```

or is there some semantic difference between `None` and `Some(BTreeMap::new())`?